entity.go

```diff
@@ -77,6 +77,34 @@ func NewMultipart(header Header, parts []*Entity) (*Entity, error) {
 	return New(header, r)
 }
 
+const maxHeaderBytes = 4000000
```

Comment:
`net/http`'s default is `1 << 20` (ie. 1MiB). I wonder what default we should pick.

Reply:
well... depends on what you aim for.

Sane headers didn't have an issue to begin with.  
The problem is things like s/mime et all that shove a public key into the headers. Or similar wacky constructs.

Maybe we can adjust this ti the default of what postfix and other smtp servers accept?

poatfix default is 

>header_size_limit (default: 102400)
The maximal amount of memory in bytes for storing a message header. If a header is larger, the excess is discarded. The limit is enforced by the cleanup(8) server.

Reply:
hm, 100kB seems awfully low... is this a per header setting as in key value pair and not header as in rfc8322?

Reply:
yup, per line

>It limits the text per header LINE.
>
>Wietse

Reply:
so then setting a size is less trivial... postfix then limits the total size of the message instead, meaning you can have a large header and small body or the other way around...

Reply:
1 to 4 MB for total header size sounds good to me.